crates/doc/src/shape/limits.rs

```diff
@@ -28,7 +28,7 @@ fn squash_location_inner(shape: &mut Shape, name: &Token) {
         // Squashing of `additional*` fields is not possible here because we don't
         // have access to the parent shape
         Token::NextIndex => unreachable!(),
-        Token::Property(prop) if prop == "*" => unreachable!(),
+        Token::NextProperty => unreachable!(),
```

Comment:
nit: collapse cases. i'm harping on this only for consistency: why are we doing it in some places and not others?